modules/rte_mis_core/src/Plugin/Block/RoleBasedDetailsBlock.php

```diff
@@ -0,0 +1,474 @@
+          $termName[] = $term->label();
+        }
+        $query = $this->entityTypeManager->getStorage('user')
+          ->getQuery()
+          ->condition('roles', 'school')
+          ->condition('name', $termName, 'IN')
+          ->condition('status', 1)
+          ->accessCheck(FALSE);
+
+        $school_ids = $query->execute();
+
+        if (!empty($school_ids)) {
+          // Filter schools based on status key.
+          $verification_status = 'school_registration_verification_' . $status_key;
+          $pending_count = $this->countPendingSchools($school_ids, $verification_status);
+        }
+
+        break;
+
+      default:
+        // Default return if $current_role doesn't match any condition.
+        $pending_count = 0;
+        break;
+    }
+
+    return $pending_count;
+  }
+
+  /**
+   * Counts the number of pending schools based on given conditions.
+   *
+   * @param array $school_ids
+   *   Array of school user IDs to filter.
+   * @param string $verification_status
+   *   Verification status to filter by ('school_registration_verification_appro
+   *   ved_by_beo' or 'school_registration_verification_submitted').
+   *
+   * @return int
+   *   The count of pending schools.
+   */
+  private function countPendingSchools(array $school_ids, string $verification_status): int {
+    // Query to count pending schools.
+    $query = $this->entityTypeManager->getStorage('user')
+      ->getQuery()
+      ->condition('uid', $school_ids, 'IN')
+      ->accessCheck(FALSE)
+      ->condition('field_school_details.entity:mini_node.field_school_verification', $verification_status);
+
+    // Execute the query to get pending school user IDs.
+    $pending_nids = $query->execute();
+
+    // Count the pending schools.
+    return count($pending_nids);
+  }
+
+  /**
+   * Gets the IDs of schools matching the current user's location.
+   *
+   * This function retrieves the IDs of taxonomy terms (schools) that match
+   * the location of the current user based on their 'field_location_details'
+   * field.
+   *
+   * @return array
+   *   An array of school IDs matching the current user's location.
+   */
+  public function gettingMatchingSchoolTerms() {
+    // Get the current user's ID and load the user entity.
+    $currentUserId = $this->currentUser->id();
+
+    /** @var \Drupal\user\Entity\User */
+    $currentUser = $this->entityTypeManager->getStorage('user')->load($currentUserId);
+    $currentUserRole = $this->currentUser->getRoles(TRUE);
+
+    /** @var \Drupal\taxonomy\TermStorage */
+    $term_storage = $this->entityTypeManager->getStorage('taxonomy_term');
+    // Extract the location ID from the user's field.
+    $locationId = (int) $currentUser->get('field_location_details')->getString();
+
+    if (in_array('district_admin', $currentUserRole)) {
+      // For district get the child location.
+      $temp = $term_storage->loadTree('location', $locationId, 1, FALSE);
+      $locationId = reset($temp)->tid;
```

Comment:
Please add a check before accessing the entity.

Reply:
ok